docs/sources/query/logcli/logcli-tutorial.md

```diff
@@ -0,0 +1,464 @@
+
+### Checking series cardinality
+
+One of the most important aspects of keeping Loki healthy is to monitor the series cardinality. This is the number of unique series in your logs. A high series cardinality can lead to performance issues and increased storage costs. We can use LogCLI to check the series cardinality of our logs.
+
+To start let's print how many unique series we have in our logs:
+
+```bash
+logcli series '{}'
+```
+
+This will return a list of all the unique series in our logs. The output will look similar to the following:
+
+```console
+{package_size="Small", service_name="Delivery World", state="Florida"}
+{package_size="Medium", service_name="Delivery World", state="Florida"}
+{package_size="Small", service_name="Delivery World", state="California"}
+{package_size="Large", service_name="Delivery World", state="New York"}
+{package_size="Small", service_name="Delivery World", state="Illinois"}
+{package_size="Large", service_name="Delivery World", state="Florida"}
+{package_size="Medium", service_name="Delivery World", state="Illinois"}
+{package_size="Large", service_name="Delivery World", state="Texas"}
+{package_size="Medium", service_name="Delivery World", state="California"}
+{package_size="Medium", service_name="Delivery World", state="Texas"}
+{package_size="Small", service_name="Delivery World", state="Texas"}
+{package_size="Large", service_name="Delivery World", state="Illinois"}
+{package_size="Small", service_name="Delivery World", state="New York"}
+{package_size="Medium", service_name="Delivery World", state="New York"}
+{package_size="Large", service_name="Delivery World", state="California"}
+```
+
+We can further improve this query by adding `--analyze-labels`:
+```bash
+logcli series '{}' --analyze-labels
+```
+
+This will return a summary of the unique values for each label in our logs. The output will look similar to the following:
+
+```console
+Label Name    Unique Values  Found In Streams
+state         5              15
+package_size  3              15
+service_name  1              15
+```
+
+### Detected Fields
+
+Another useful feature of LogCLI is the ability to detect fields in your logs. This can be useful for understanding the structure of your logs and the keys that are present. This will let us detect keys which could be promoted to labels or to structured metadata. 
+
+```bash
+logcli detected-fields --since 24h '{service_name="Delivery World"}'
+```
+
+This will return a list of all the keys detected in our logs. The output will look similar to the following:
+
+```console
```

Comment:
Some higher cardinality data for ya:

label: city                                  type: string               cardinality: 15
label: detected_level              type: string               cardinality: 3
label: note                              type: string               cardinality: 7
label: package_id                  type: string               cardinality: 994
label: package_size_extracted              type: string               cardinality: 3
label: package_status           type: string            cardinality: 4
label: package_type             type: string            cardinality: 5
label: receiver_address         type: string            cardinality: 991
label: receiver_name            type: string            cardinality: 100
label: sender_address           type: string            cardinality: 991
label: sender_name              type: string            cardinality: 100
label: state_extracted          type: string            cardinality: 5
label: timestamp                type: string            cardinality: 1000